sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -240,19 +240,27 @@ object GpuDeviceManager extends Logging {
         (conf.rmmAllocFraction * (info.free - reserveAmount)).toLong)
       if (poolAllocation < minAllocation) {
         throw new IllegalArgumentException(s"The pool allocation of " +
-            s"${toMB(poolAllocation)} MB (calculated from ${RapidsConf.RMM_ALLOC_FRACTION} " +
-            s"(=${conf.rmmAllocFraction}) and ${toMB(info.free)} MB free memory) was less than " +
-            s"the minimum allocation of ${toMB(minAllocation)} (calculated from " +
-            s"${RapidsConf.RMM_ALLOC_MIN_FRACTION} (=${conf.rmmAllocMinFraction}) " +
-            s"and ${toMB(info.total)} MB total memory)")
+            s"${toMB(poolAllocation)} MiB (gpu.free: ${toMB(info.free)}," +
+            s"${RapidsConf.RMM_ALLOC_FRACTION}: (=${conf.rmmAllocFraction}," +
+            s"${RapidsConf.RMM_ALLOC_RESERVE}: ${reserveAmount} => " +
+            s"(gpu.free - reserve) * allocFraction = ${toMB(poolAllocation)})" +
+            s"was less than allocation of ${toMB(minAllocation)} MiB (gpu.total: " +
```

Comment:
I think we can bring this message into two: the part before "was less than" and the part after. That way we can reuse it in the second case (it is almost verbatim except for this text, as far as I can tell).

Reply:
agree